src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -436,6 +531,176 @@ CreateDistributedTable(Oid relationId, Var *distributionColumn, char distributio
 }
 
 
+/*
+ * CreateCitusLocalTable is the internal method to create citus local table's
+ * shard and metadata.
+ * Note that this function does not perform any validation on the table as it
+ * is already done in create_citus_local_table.
+ */
+void
+CreateCitusLocalTable(Oid relationOid)
+{
+	/*
+	 * Make sure that existing reference tables have been replicated to all
+	 * the nodes such that we can create foreign keys and joins work immediately
+	 * after creation.
+	 */
+	EnsureReferenceTablesExistOnAllNodes();
+
+	/*
+	 * Get an exclusive lock on relation with relationOid the operations done
+	 * in this function should be automic.
+	 */
+	LockRelationOid(relationOid, ExclusiveLock);
+
+	/*
+	 * Get necessary commands to re-create the shell table before renaming the
+	 * given relation to the shard relation.
+	 */
+	List *foreignConstraintCommands =
+		GetForeignConstraintCommandsTableInvolved(relationOid);
+
+	List *triggerCommands = GetTableTriggerCommands(relationOid);
+
+	/* include DEFAULT clauses for columns getting their default values from a sequence */
+	bool includeSequenceDefaults = true;
+
+	/*
+	 * Do not include foreign table dependencies as we are already creating the
+	 * citus local table in the current node
+	 */
+	bool includeForeignTableDependencies = false;
+
+	List *tableDDLEvents = GetTableDDLEvents(relationOid, includeSequenceDefaults,
+											 includeForeignTableDependencies);
+	tableDDLEvents = list_concat(tableDDLEvents, foreignConstraintCommands);
+	tableDDLEvents = list_concat(tableDDLEvents, triggerCommands);
+
+	uint64 shardId = GetNextShardId();
+
+	char *relationName = get_rel_name(relationOid);
```

Comment:
relationOid/relationName  are generic to be used here. Can you give  explicit variable names? Like `shellRelationOid` / `shardRelationOid` etc